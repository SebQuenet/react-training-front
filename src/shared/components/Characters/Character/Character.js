// @flow

import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles({
  card: {
    minWidth: 250,
    marginBottom: 12,
    marginRight: 12,
    marginLeft: 12,
    marginTop: 12,
  },
  bullet: {
    display: 'inline-block',
    margin: '0 2px',
    transform: 'scale(0.8)',
  },
  title: {
    fontSize: 14,
  },
  pos: {
    marginBottom: 12,
  },
});

type PropTypes = {
  character: Object,
};

/**
 * Character component
 * @param {Object} character
 */
const Character = ({ character }: PropTypes) => {
  const classes = useStyles();
  const notDefined = 'Non Defini(e)';

  return (
    <Card className={classes.card}>
      <CardContent>
        <Typography className={classes.title} color="textSecondary" gutterBottom>
          Acteur
        </Typography>
        {character.playedBy.map((play, index) => (
          <Typography key={index} variant="h5" component="h2">
            {play !== '' ? play : notDefined}
          </Typography>
        ))}
        <Typography className={classes.pos} color="textSecondary">
          RÃ´le(s)
        </Typography>
        {character.aliases.map((alias, index) => (
          <Typography key={index} variant="body2" component="p">
            {alias}
          </Typography>
        ))}
      </CardContent>
    </Card>
  );
};

export default Character;
