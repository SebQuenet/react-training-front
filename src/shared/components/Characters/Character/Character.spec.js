import React from 'react';
import renderer from 'react-test-renderer';
import _cloneDeep from 'lodash/cloneDeep';
import Character from './Character';

const defaultProps = {
  character: {
    url: 'https://anapioficeandfire.com/api/characters/2',
    name: 'Walder',
    gender: 'Male',
    culture: '',
    born: '',
    died: '',
    titles: [''],
    aliases: [
      'Balon the Brave',
      'Balon the Blessed',
      'Balon the Twice Crowned',
      'Balon the Widowmaker',
      'The Kraken King',
    ],
    father: '',
    mother: '',
    spouse: '',
    allegiances: ['https://anapioficeandfire.com/api/houses/362'],
    books: [
      'https://anapioficeandfire.com/api/books/1',
      'https://anapioficeandfire.com/api/books/2',
      'https://anapioficeandfire.com/api/books/3',
      'https://anapioficeandfire.com/api/books/5',
      'https://anapioficeandfire.com/api/books/8',
    ],
    povBooks: [],
    tvSeries: ['Season 1', 'Season 2', 'Season 3', 'Season 4', 'Season 6'],
    playedBy: ['Patrick Malahide'],
  },
};

describe('Character component', () => {
  it('should render correctly when the real name of actor is define', () => {
    // given
    const component = renderer.create(<Character {...defaultProps} />);
    // when
    const tree = component.toJSON();
    // then
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when the real name of actor is not define', () => {
    // given
    const props = _cloneDeep(defaultProps);
    props.character.playedBy[0] = '';
    const component = renderer.create(<Character {...props} />);
    // when
    const tree = component.toJSON();
    // then
    expect(tree).toMatchSnapshot();
  });
});
