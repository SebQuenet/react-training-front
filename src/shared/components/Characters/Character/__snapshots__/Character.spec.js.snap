// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component should render correctly when the real name of actor is define 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root makeStyles-title-3 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
    >
      Acteur
    </p>
    <h2
      className="MuiTypography-root MuiTypography-h5"
    >
      Patrick Malahide
    </h2>
    <p
      className="MuiTypography-root makeStyles-pos-4 MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      Rôle(s)
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
      Balon the Brave
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
      Balon the Blessed
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
      Balon the Twice Crowned
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
      Balon the Widowmaker
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
      The Kraken King
    </p>
  </div>
</div>
`;

exports[`Character component should render correctly when the real name of actor is not define 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root makeStyles-title-3 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
    >
      Acteur
    </p>
    <h2
      className="MuiTypography-root MuiTypography-h5"
    >
      Non Defini(e)
    </h2>
    <p
      className="MuiTypography-root makeStyles-pos-4 MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      Rôle(s)
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
      Balon the Brave
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
      Balon the Blessed
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
      Balon the Twice Crowned
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
      Balon the Widowmaker
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
      The Kraken King
    </p>
  </div>
</div>
`;
